frontend/src/components/generic/SocialLinks/SocialIcons.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faBriefcase, faEnvelope } from '@fortawesome/free-solid-svg-icons'
+import {
+    faGithub,
+    faLinkedin,
+    faDiscord,
+    faSlack,
+} from '@fortawesome/free-brands-svg-icons'
+import junctionStyle from 'utils/styles'
+import { popupCenter } from 'utils/misc'
+
+const SocialIcons = ({ social, email = null }) => {
+    const classes = junctionStyle()
+
+    const socials = [
+        social.github && { icon: faGithub, link: social.github },
+        social.linkedin && { icon: faLinkedin, link: social.linkedin },
+        social.slack && { icon: faDiscord, link: social.slack },
+        social.discord && { icon: faSlack, link: social.discord },
+        social.portfolio && {
+            icon: faBriefcase,
+            link: social.portfolio,
+        },
+        email && {
+            icon: faEnvelope,
+            link: `mailto:${email}`,
+        },
+    ]
```

Comment:
- [ ] Required: This code fails if the social prop being passed doesn't contain one of the elements in the array
- [ ] Required: The list of possible icons is missing the Telegram icon